services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DelegatingDeviceManagementHttpEndpoint.java

```diff
@@ -244,4 +285,23 @@ private void doDeleteDevice(final RoutingContext ctx) {
                     () -> result.complete(new Device()));
         return result.future();
     }
+
+    private <T> Future<Optional<List<T>>> decodeJsonFromRequestParameter(final RoutingContext ctx,
+            final String paramKey, final Class<T> clazz) {
+
+        Objects.requireNonNull(ctx);
+        Objects.requireNonNull(paramKey);
+        Objects.requireNonNull(clazz);
+
+        final Promise<Optional<List<T>>> result = Promise.promise();
+        final Optional<List<T>> values = Optional.ofNullable(ctx.request().params()
+                .getAll(paramKey))
+                .map(jsons -> jsons
+                        .stream()
+                        .map(json -> Json.decodeValue(json, clazz))
+                        .collect(Collectors.toList()));
+        result.complete(values);
+
+        return result.future();
```

Comment:
how about failing the result with a ClientErrorExecption in case the JSON cannot be parsed?